operator/pkg/clusterconfiguration/cel_patcher_test.go

```diff
@@ -0,0 +1,173 @@
+package clusterconfiguration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/ptr"
+)
+
+func TestCelPatcher(t *testing.T) {
+	redpanda := config.RedpandaYaml{
+		Redpanda: config.RedpandaNodeConfig{
+			Directory:     "/foo/bar",
+			DeveloperMode: true,
+		},
+		Rpk: config.RpkNodeConfig{
+			Tuners: config.RpkNodeTuners{
+				TuneNetwork:       true,
+				TuneDiskScheduler: false,
+			},
+		},
+		Pandaproxy: nil,
+		PandaproxyClient: &config.KafkaClient{
+			Brokers: []config.SocketAddress{{
+				Address: "0.0.0.0",
+				Port:    9092,
+			}},
+			SASLMechanism: ptr.To("SCRAM-SHA-256"),
+			SCRAMUsername: ptr.To("user"),
+			SCRAMPassword: nil,
+		},
+		SchemaRegistry:       nil,
+		SchemaRegistryClient: nil,
+		Other:                nil,
+	}
+
+	environ := map[string]string{
+		"env1": "bar",
+	}
+	factory := StdLibFactory(environ)
+
+	tpl := Template[*config.RedpandaYaml]{
+		Content: &redpanda,
+		Fixups: []Fixup{
+			// {Field: "redpanda", CEL: "true"},
+			{Field: "redpanda.data_directory", CEL: `"updated directory"`},
+			{Field: "pandaproxy_client.scram_username", CEL: `it + "...blah"`},
+			{Field: "pandaproxy_client.scram_password", CEL: `"foo" + envString("env1")`},
```

Comment:
Fix ups would need to be applied by both the configurator and the operator, right? If so, I think we'd need/want a `secretKey("secret name", "key")` function which is defined differently depending on the context. (Though we could technically give the configurator k8s API access 🤔 )

Reply:
At the moment I'm ensuring that these are env-var available. The alternative is indeed to give the container the ability to get k8s secret data directly.

Reply:
Which is to say: at the moment the CEL stuff is an under-the-hood implementation approach. If we want to expose it as CRD fields then having higher-level functions like `secretKey` would be useful, but I wanted to start by keeping it hidden from that layer (so the details can be flexible for a time).